### PR TITLE
Introduce React components library and configure monorepo

### PR DESCRIPTION
**Pull request by pkosiec**
## Description

Changes proposed in this pull request:

- Prepare Lerna configuration
- Prepare components setup for testing and development live-reload
- Add stylebook to showcase React components
- Use global ESLint configuration
- Add styled-components for both Dashboard and Components library

## Known issues

Race condition while running `npm start` dev setup for all packages. Sometimes when you update a react component, Dashboard refreshes, but it still takes old files of react components to the build, which means sometimes you need to do a second change in the library to see updated

Probably this is the property which could fix that: https://webpack.js.org/configuration/watch/#watchoptionsaggregatetimeout

I tried to fix that with:
- using `rewire` - unfortunately we would need to override https://github.com/facebook/create-react-app/blob/v5.0.0/packages/react-scripts/config/webpack.config.js#L100, which in this case would mean almost 1:1 copy of this file
- enabling polling (with `CHOKIDAR_USEPOLLING` and `CHOKIDAR_INTERVAL`) - but as I wrote above, it is mostly related to the `aggregateTimeout` on webpack side, so...

Unfortunately I ran out of time, so I will describe it as a separate GH issue. I think this is not that big deal, as you can still develop React components or Dashboard separately. or accept that sometimes you need to do a second change in the codebase to refresh the build.
Also, maybe the issue will disappear once we will cross the 200ms build for react components (that is: we move most components there) 🤷 

I will also mention this issue in main Readme in the following PR.

## Testing

1. Follow the instructions in the main readme to run the development setup both for Dashboard and its dependencies.
2. Navigate to `react-components` and run `npm run storybook` to see the Components explorer UI. 

### CI

Branch: https://github.com/pkosiec/dashboard/runs/5130121568?check_suite_focus=true
PR: https://github.com/pkosiec/dashboard/actions/runs/1819800549

## Related issue(s)

#21

### CODE REVIEW

File: .github/workflows/branch-build.yaml [hunk 1]
```diff
@@ -37,12 +32,14 @@ jobs:
           cache-dependency-path: '**/package-lock.json'
 
       - name: Install dependencies
-        run: npm install
+        run: |
+          npm install
+          npm run bootstrap
 
       - name: Lint code
```
Comment (mkuziemko) [inline .github/workflows/branch-build.yaml:39]: Should we run not only linting but also executing the tests? (even if there are no test files yet)

File: package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "prettier": "^2.5.0"
+  },
+  "scripts": {
+    "postinstall": "if [ \"${SKIP_POSTINSTALL}\" != \"true\" ]; then npm run setup-hooks; else echo \"Post install script skipped as the env SKIP_POSTINSTALL is set to true.\"; fi",
+    "setup-hooks": "git config core.hooksPath ./scripts/git-hooks",
+    "bootstrap": "lerna bootstrap",
+    "clean": "lerna clean",
+    "start": "lerna run start --parallel",
+    "lint": "eslint . -c ./.eslintrc.js",
+    "test": "lerna run test",
```
Comment (mkuziemko) [inline package.json:15]: how about adding `--parallel` flag?

File: dashboard/src/auth/LoginForm.tsx [hunk 3]
```diff
@@ -10,26 +11,41 @@ interface LoginFormProps {
...
   return (
-    <div className="login-form">
-      <Card
-        title="Capact Dashboard Login"
-        bordered={false}
-        style={{ width: 500 }}
-      >
+    <LoginFormWrapper>
+      <FixedWidthCard title="Capact Dashboard Login" bordered={false}>
+        {/* TODO: Just for testing purposes - remove before merge */}
```
Comment (mkuziemko) [inline dashboard/src/auth/LoginForm.tsx:41]: `todos`

File: react-components/package.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
+{
```
Comment (mkuziemko) [inline react-components/package.json:1]: should we also add here some `license`?

**Review by mkuziemko — COMMENTED** [review]
Tested it locally and it works great! :+1: Small comments/questions added.

File: .github/workflows/branch-build.yaml (hunk 1 above)
Comment (pkosiec) [reply .github/workflows/branch-build.yaml:39]: Good point 👍

**Comment by mszostok** [comment]
I didn't check this PR, but I spot two problems on my two other PRs:
- ` npm run setup-hooks` doesn't work
- the `build` is created on the root and it's not ignored


Please just ensure that it's fixed :) If it's already fixed then ignore this comment :)

**Comment by pkosiec** [comment]
Yep, @mszostok These are already fixed in one of my PRs 🙂  The same with outdated Policy GraphQL query (after nesting all properties under `rules`).

**Comment by mszostok** [comment]
@pkosiec could you tell me in which? So I will close this one: https://github.com/capactio/dashboard/pull/25

**Comment by pkosiec** [comment]
@mszostok Sure - on #24 - https://github.com/capactio/dashboard/pull/24/files#diff-c67c98d7ff604d453441c65e72ae2fa2b1f4a6dd0278cc9f2fa28ac711fcdd82

**Review by mkuziemko — APPROVED** [review]
LGTM :+1:

**Pull request merged**